### PR TITLE
refactor(Scripts/Northrend): code cleanup (part 11)

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
- Just a minor code cleanup, follow up of https://github.com/azerothcore/azerothcore-wotlk/pull/7065
- Cleaning `src/scripts/Northrend/**` source files

### CODE REVIEW

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_razorscale.cpp [hunk 1]
```diff
@@ -208,16 +207,22 @@ class boss_razorscale : public CreatureScript
...
+                            spellId = SPELL_CHAIN_1;
+                        }
+                        else if (caster->GetGUID() == pInstance->GetGuidData(DATA_HARPOON_FIRE_STATE_2))
+                        {
+                            spellId = SPELL_CHAIN_2;
+                        }
+                        else if (caster->GetGUID() == pInstance->GetGuidData(DATA_HARPOON_FIRE_STATE_3))
+                        {
+                            spellId = SPELL_CHAIN_3;
+                        }
```
Comment (Helias) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_razorscale.cpp:223]: we could convert this into switch case or at least call caster->GetGUID() only once saving it in a variable

Comment (Helias) [inline src/server/scripts/Northrend/Ulduar/Ulduar/boss_razorscale.cpp:223]:
```suggestion
                        switch(caster->GetGUID()) {
                            case pInstance->GetGuidData(DATA_HARPOON_FIRE_STATE_1:
                            {
                                spellId = SPELL_CHAIN_1;
                                break;
                            }
                            case pInstance->GetGuidData(DATA_HARPOON_FIRE_STATE_2:
                            {
                                spellId = SPELL_CHAIN_2;
                                break;
                            }
                            case pInstance->GetGuidData(DATA_HARPOON_FIRE_STATE_3:
                            {
                                spellId = SPELL_CHAIN_3;
                                break;
                            }
                        }
```

File: src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp [hunk 2]
```diff
@@ -704,11 +704,14 @@ class instance_ulduar : public InstanceMapScript
                     {
                         instance->LoadGrid(364.0f, -16.0f); //make sure leviathan is loaded
                         m_leviathanTowers[type - EVENT_TOWER_OF_LIFE_DESTROYED] = data;
-                        GameObject* gobj = nullptr;
-                        if ((gobj = instance->GetGameObject(m_leviathanVisualTowers[type - EVENT_TOWER_OF_LIFE_DESTROYED][0])))
-                            gobj->SetGoState(GO_STATE_ACTIVE);
-                        if ((gobj = instance->GetGameObject(m_leviathanVisualTowers[type - EVENT_TOWER_OF_LIFE_DESTROYED][1])))
-                            gobj->SetGoState(GO_STATE_ACTIVE);
+                        if (GameObject* gameObject = instance->GetGameObject(m_leviathanVisualTowers[type - EVENT_TOWER_OF_LIFE_DESTROYED][0]))
```
Comment (lineagedr) [inline src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp:707]: How about?
```c++
for (uint8 i = 0; i < 2; ++i)
    if (GameObject* gameObject = instance->GetGameObject(m_leviathanVisualTowers[type - EVENT_TOWER_OF_LIFE_DESTROYED][i]))
        gameObject->SetGoState(GO_STATE_ACTIVE);
```

File: src/server/scripts/Northrend/zone_dragonblight.cpp [hunk 3]
```diff
@@ -1620,20 +1619,18 @@ class npc_commander_eligor_dawnbringer : public CreatureScript
...
                 std::list<Creature*> creatureList;
                 GetCreatureListWithEntryInGrid(creatureList, me, AudienceMobs[ii], 15.0f);
                 for (std::list<Creature*>::iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
                 {
-                    if (Creature* creatureList = *itr)
+                    if (Creature* creature = *itr)
                     {
-                        audienceList[creaturecount] = creatureList->GetGUID();
-                        ++creaturecount;
+                        audienceList[creturesCount] = creature->GetGUID();
```
Comment (lineagedr) [inline src/server/scripts/Northrend/zone_dragonblight.cpp:1632]: Do we  need creturesCount? we can use ii it gets incremented the same way

```c++
audienceList[ii] = creature->GetGUID();
```

File: src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp (hunk 2 above)
Comment (FrancescoBorzi) [reply src/server/scripts/Northrend/Ulduar/Ulduar/instance_ulduar.cpp]: done, thanks for the tip!

**Review by Helias — APPROVED** [review]

**Pull request merged**